utils/verify-package.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+set -euo pipefail
+
+# everything merged to 'main' is published to this URL
+packageUrl="https://api-frameworks.zeit.sh/tarballs/vercel.tgz"
```

Comment:
I think we need to wait for the url from the specific commit because this script might run before the Vercel deployment completes

Reply:
Ah, sounds like we could really use: https://linear.app/vercel/issue/VCCLI-261/create-cli-command-to-wait-for-deployment

Reply:
@EndangeredMassa We have that already

https://github.com/vercel/vercel/blob/aaceeef604ffe432af15d5f8f315c8ebc5db98f5/.github/workflows/test.yml#L23-L27

Reply:
I'm actually not even sure if we can catch this category of bug because we don't know what `lerna` will do. So we kinda have to test against the published package.

Perhaps a better solution is to update CI to include a rollback mechanism.

Basically publish, try npm install, then npm dist-tag to previous version if that failed to install

Reply:
As is, this PR is more protection than we had before. I think it's mergable, but we should keep in mind that the latest main CI has to have passed (and that deployment needs to be ready) before we can publish with certainty.

We can revise this by waiting for that deployment after making the CLI command for it.

Reply:
How will we know which tarball was tested? Might give a false sense of security right now.

Reply:
Great, thanks for verifying my theory 👍 

I do think the tarball test could be somewhat useful for running against every `main` commit and maybe catch a different class of bug earlier (perhaps turbo or ncc misbehaving)